:rotating_light: NIGHTLY DEPENDENCIES TEST: failure on `refs/heads/main`

The run `12199328751` of the workflow testing Nilearn with the nightly build of its dependencies failed on `refs/heads/main`.

You can view the report here: https://github.com/nilearn/nilearn/actions/runs/12199328751

Pinging @nilearn/core

looks like more sklearn tags shenanigans:

```
FAILED nilearn/maskers/tests/test_base_masker.py::test_check_estimator_invalid[estimator4-check4-check_positive_only_tag_during_fit]
FAILED nilearn/regions/tests/test_hierarchical_kmeans_clustering.py::test_check_estimator_invalid[estimator2-check2-check_positive_only_tag_during_fit]
```

--- a/nilearn/maskers/tests/test_base_masker.py
+++ b/nilearn/maskers/tests/test_base_masker.py
@@ -31,6 +31,9 @@ extra_valid_checks = [
 if compare_version(sklearn_version, "<", "1.5.0"):
     extra_valid_checks.append("check_estimator_sparse_data")
 
+if compare_version(sklearn_version, ">", "1.6"):
+    extra_valid_checks.append("check_positive_only_tag_during_fit")
+
 
 @pytest.mark.parametrize(
     "estimator, check, name",

--- a/nilearn/regions/tests/test_hierarchical_kmeans_clustering.py
+++ b/nilearn/regions/tests/test_hierarchical_kmeans_clustering.py
@@ -50,6 +50,9 @@ if compare_version(sklearn_version, ">", "1.5.2"):
 if compare_version(sklearn_version, "<", "1.5.0"):
     extra_valid_checks.append("check_estimator_sparse_data")
 
+if compare_version(sklearn_version, ">", "1.6"):
+    extra_valid_checks.append("check_positive_only_tag_during_fit")
+
 
 @pytest.mark.parametrize(
     "estimator, check, name",

